maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/SpaInstaller.java

```diff
@@ -101,18 +105,21 @@ public void installFromDistroProperties(File appDataDir, DistroProperties distro
 
 		// First pull any optional properties that may be used to specify the core, node, or npm versions
 		// These properties are not passed to the build tool, but are used to specify the build execution itself
-		String coreVersion = spaBuildProperties.remove("core");
-		if (coreVersion == null) {
-			coreVersion = "next";
-		}
-		String nodeVersion = spaBuildProperties.remove("node");
-		if (nodeVersion == null) {
-			nodeVersion = NODE_VERSION;
-		}
-		String npmVersion = spaBuildProperties.remove("npm");
-		if (npmVersion == null) {
-			npmVersion = NPM_VERSION;
-		}
+		// Use defaults but allow overrides from distro.properties, system properties, or environment variables
+		String coreVersion = Optional.ofNullable(spaBuildProperties.remove("core"))
+				.orElse("next");
+
+		String nodeVersion = Optional.ofNullable(spaBuildProperties.remove("node.version"))
+				.orElseGet(() -> System.getProperty("sdk.node.version",
+						System.getenv().getOrDefault("SDK_NODE_VERSION", NODE_VERSION)));
+
+		String npmVersion = Optional.ofNullable(spaBuildProperties.remove("npm.version"))
```

Comment:
Same here.  If it was just `npm` previously, then it should not be changed to `npm.version`.

Reply:
So, i have added a fallback for backwards compatibility. Now we can support both `spa.npm` and `spa.npm.version`